ospd_openvas/openvas_db.py

```diff
@@ -60,219 +44,234 @@
+        if self.rediscontext is None:
+            print("Problem retrieving Redis Context")
+            return 2
+
+        return self.rediscontext
+
+    def select_kb(self, ctx, kbindex, set_global=False):
+        """ Use an existent redis connection and select a redis kb.
+        If needed, set the ctx as global.
+        Parameters:
+            ctx (redis obj): Redis context to use.
+            newdb (str):  The new kb to select
+            set_global (bool, optional): If should be the global context.
+        """
+        ctx.execute_command('SELECT '+ str(kbindex))
+        if set_global:
+            self.set_redisctx(ctx)
+
+    def item_get_list(self, name):
+        """ Get all values under a KB key list.
+        The right rediscontext must be already set.
+        """
+        ctx = self.get_kb_context()
+        return ctx.lrange(name, 0, -1)
+
+    def remove_list_item(self, key, value):
+        """ Remove item from the key list.
+        The right rediscontext must be already set.
+        """
+        ctx = self.get_kb_context()
+        ctx.lrem(key, 0, value)
+
+    def item_get_single(self, name):
+        """ Get a single KB element. The right rediscontext must be
+        already set.
+        """
+        ctx = self.get_kb_context()
+        return ctx.lindex(name, 0)
+
+    def item_add_single(self, name, values):
+        """ Add a single KB element with one or more values.
+        The right rediscontext must be already set.
+        """
+        ctx = self.get_kb_context()
+        ctx.rpush(name, *set(values))
+
+    def item_set_single(self, name, value):
+        """ Set (replace) a new single KB element. The right
+        rediscontext must be already set.
+        """
+        ctx = self.get_kb_context()
+        pipe = ctx.pipeline()
+        pipe.delete(name)
+        pipe.rpush(name, *set(value))
+        pipe.execute()
+
+    def item_del_single(self, name):
+        """ Delete a single KB element. The right rediscontext must be
+        already set.
+        """
+        ctx = self.get_kb_context()
+        ctx.delete(name)
+
+    def get_pattern(self, pattern):
+        """ Get all items stored under a given pattern.
+        """
+        ctx = self.get_kb_context()
+        items = ctx.keys(pattern)
+
+        elem_list = []
+        for item in items:
+            elem_list.append([item, ctx.lrange(item, 0, -1)])
+        return elem_list
+
+    def get_elem_pattern_by_index(self, pattern, index=1):
+        """ Get all items with index 'index', stored under
+        a given pattern.
+        """
+        ctx = self.get_kb_context()
+        items = ctx.keys(pattern)
+
+        elem_list = []
+        for item in items:
+            elem_list.append([item, ctx.lindex(item, index)])
+        return elem_list
+
+    def release_db(self, kbindex=0):
+        """ Connect to redis and select the db by index.
+        Flush db and delete the index from dbindex_name list.
+        """
+        if kbindex:
```

Comment:
Personally I would switch the logic here but that's nitpicking
```
if not kbindex:
    return
```

Btw. currently it won't go into the following line for the default case with kbindex=0. Is this valid? or should it be if kbindex is not None?